matching-notification-api/src/main/java/org/phenotips/matchingnotification/internal/DefaultMatchingNotificationManager.java

```diff
@@ -193,12 +198,15 @@ private void filterNonUsersMatches(List<PatientMatch> matches)
     {
         ListIterator<PatientMatch> iterator = matches.listIterator();
         while (iterator.hasNext()) {
-            PatientMatch match = iterator.next();
-            boolean hasViewAccess = this.viewAccess.compareTo(match.getReference().getAccess()) <= 0
-                && this.viewAccess.compareTo(match.getMatched().getAccess()) <= 0;
```

Comment:
@veronikaslc I'm not sure why we used to have a `&&` here. That would imply you have view access to both reference and matched patients, which is basically never the case in matching notification which always shouws a match between one of your patients and one of someone else's. It sued to work because I think only admin users went through this code (and they have access to everything), but please verify my logic to replace with with a `||`

Reply:
Yup, seems `&&` was a bug, should be `||`, also confirmed by this code https://github.com/phenotips/patient-network/blob/stable-1.1.10-rc-2/matching-notification-api/src/main/java/org/phenotips/matchingnotification/storage/internal/DefaultMatchStorageManager.java#L324-L329